openerp-client/src/main/java/org/bahmni/openerp/web/http/client/RestClient.java

```diff
@@ -31,28 +32,25 @@ public RestClient(String baseURL, String username, String password, int connecti
     }
 
     private void login() {
```

Comment:
I would refactor the code to say: 

if(sessionIsInvalid)
  performOdooLoginWithSystemUser()
else
  checkOdooLoginWorks()

And extract two different methods called: performOdooLoginWithSystemUser(), and checkOdooLoginWorks()

This will help make the login() method more readable.